internal/server/handlers.go

```diff
@@ -289,6 +297,19 @@ func (h *handler) StartNotificationBatch(initialResourceVersions map[string]stri
 		for name, version := range initialResourceVersions {
 			h.initialResourceVersions[name] = &initialResourceVersion{version: version}
 		}
+	} else if estimatedSize > 0 {
+		// Only preallocate the send buffer if the initialResourceVersions is empty. Otherwise, it's very
```

Comment:
> Only preallocate the send buffer if the initialResourceVersions is empty

This seems not true in subscription_manager.go, where `len(initialResourceVersions) > 0`, see my other comment there. You may want to move this comment to there, since the check on initialResourceVersions is over there.